google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java

```diff
@@ -61,32 +63,42 @@ public void deploy(@NotNull final DeploymentTask<AppEngineDeploymentConfiguratio
       return;
     }
 
-    AppEngineHelper appEngineHelper = new CloudSdkAppEngineHelper(task.getProject());
-
     AppEngineDeploymentConfiguration deploymentConfig = task.getConfiguration();
 
-    final CancellableRunnable deployRunner =  appEngineHelper.createDeployRunner(
-        logManager.getMainLoggingHandler(),
-        task.getSource(),
-        deploymentConfig,
-        callback);
+    AppEngineEnvironment environment =
+        AppEngineEnvironment.valueOf(deploymentConfig.getEnvironment());
+    if (environment == AppEngineEnvironment.APP_ENGINE_FLEX
+        && Strings.isNullOrEmpty(deploymentConfig.getYamlPath())) {
+      callback.errorOccurred(GctBundle.getString("appengine.deployment.error.staging.yaml")
+          + "\n" + GctBundle.getString("appengine.deployment.error.suggestflexfacet"));
```

Comment:
Users with run configurations to deploy to Flex that won't be able to run the run configurations again after updating the plugin, because they won't see the deployables, because they haven't added the Flex facet.